### PR TITLE
Upgrade FastAPI to 0.87.0, drop old workarounds

### PR DESCRIPTION
**Pull request by mvdbeek**
## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by nsoranzo** [comment]
Thanks, can you fix the conflict please?

File: lib/galaxy/webapps/base/api.py [hunk 1]
```diff
@@ -7,14 +7,9 @@
     FastAPI,
     HTTPException,
     Request,
-    status,
```
Comment (nsoranzo) [inline lib/galaxy/webapps/base/api.py]: Bad merge?

File: test/unit/util/test_get_url.py [hunk 2]
```diff
@@ -3,6 +3,7 @@
 import responses
 
 from galaxy.util import url_get
+from galaxy_test.base.populators import skip_if_site_down
```
Comment (nsoranzo) [inline test/unit/util/test_get_url.py]: This breaks the packaging structure.
`skip_if_site_down` is actually in `lib/galaxy/tool_util/unittest_utils/__init__.py` , but that would still break the packaging structure. I think we need to create `lib/galaxy/util/unittest_utils/__init__.py` and move it there (and fix the imports across the code base).

Comment (mvdbeek) [reply test/unit/util/test_get_url.py]: `get_url` is a tool shed thing, maybe I can just move this into lib/galaxy/tool_shed/util ?

Comment (nsoranzo) [reply test/unit/util/test_get_url.py]: @jmchilton may know ^

Comment (mvdbeek) [reply test/unit/util/test_get_url.py]: it's also used in authnz (probably not necessary, but I don't want to touch everything). I don't know if I love having these unittest_utils packages everywhere ...

Comment (nsoranzo) [reply test/unit/util/test_get_url.py]: Yep, I guess it's the lesser evil :D

Forgot to `git add lib/galaxy/util/unittest_utils/__init__.py` ?

**Review by nsoranzo — APPROVED** [review]

**Comment by martenson** [comment]
just a note that this release of Starlette is what fixed it I think: https://github.com/encode/starlette/releases/tag/0.21.0

**Pull request merged**